src/NeuronNearestNeighbor.cpp

```diff
@@ -290,65 +302,149 @@ enum class mode : int
     count
 };
 
+std::string mode_to_str(mode m) {
+    std::string mode_str;
+    switch (m)
+    {
+        case mode::listed: {
+            mode_str = "listed";
+            break;
+        }
+        case mode::random_inf: {
+            mode_str = "random_inf";
+            break;
+        }
+        case mode::sin_to_matrix: {
+            mode_str = "sin_to_matrix";
+            break;
+        }
+        case mode::read_matrix: {
+            mode_str = "read_matrix";
+            break;
+        }
+        case mode::count: {
+            mode_str = "count";
+            break;
+        }
+    }
+    return mode_str;
+}
+
 #define USAGE_MSG "USAGE: ./sinblast ... followed by one of the following:\n"\
 "    -l query.swc [ list of target.swc's ] # pair the query against all listed targets, produces .sin files|\n"\
-"    -r [list of swc files] # produce random pairs, ad infinitum, produces .sin files |\n"\
+"    -r [rand_count] [list of swc files] # produce random pairs, ad infinitum if number of random pairs == -1, produces .sin files |\n"\
 "    -s [sin file] # turn a sin file into a p-value matrix, produces a .matrix file |\n"\
```

Comment:
Convention is that square brackets denote _optional_ arguments, but I'm pretty sure -r _requires_ extra args. so I think it would be
 -r N S1 [S2 ...] where N specifies the number of pairs (N=ad infinitum) and S1, S2, etc, are the input SWC files
 -s sin_file

(see for example "man cp", which has required arguments)